developer/src/kmc/src/util/extendedCompilerOptions.ts

```diff
@@ -7,4 +9,137 @@ export interface ExtendedCompilerOptions extends CompilerOptions {
    * MIT
    */
   forPublishing?: boolean;
+  /**
+   * Overrides for message reporting
+   */
+  messageOverrides?: CompilerMessageOverrideMap;
+};
+
+/**
+ * converts an --message command line parameter `<id>[:level]` and verifies that
+ * it is a valid message, returning `null` if it is not a valid id or level. If
+ * level is omitted, the message is disabled. Valid levels are Disable, Info,
+ * Hint, Warn or Error (can shorten or first character of each).
+ *
+ * Will report details of any errors to callbacks.reportMessage
+ * @param message    command line parameter in format '["KM"]<id>[":"level]'
+ * @param callbacks
+ * @returns  CompilerMessageOverride map with the new severity level for the
+ * message; or `null` if parameter is invalid.
+ */
+function commandOptionsMessageToCompilerOptionsMessage(message: string, callbacks: CompilerCallbacks): CompilerMessageOverride {
```

Comment:
It appears this method and the next are responsible for converting the command-line message options into a map used by the "focal point" noted above.

Took a while to find it; the original version (from kmc/src/commands/build.ts) had a notably different form that made me _think_ it was a minor change until I landed on the versions seen here, separately.  Would've been easier to find if the PR was more tightly-focused / split into pieces to make it clearer, but I eventually got there.